numbers/randBeta.go

```diff
@@ -0,0 +1,182 @@
+package numbers
+
+import (
+	"math"
+	"math/rand"
+	"log"
+)
+
+const RelError float64 = 1.0e-8
+const Small float64 = 1.0e-30
+const MaxIterations int = 200
+
+//RandBeta returns a random number drawn from a beta distribution with parameters alpha and beta as an xy point where y is the function density.
+//Implemented from principles in R. C. H. Cheng (1978). Generating beta variates with nonintegral shape parameters. Communications of the ACM 21, 317-322.
+func RandBeta(alpha float64, beta float64) float64 {
+	r := rand.Float64()
+	for r <= 0.0 || r >= 1.0 {//catches the case where r is exactly  0.0 or 1.0
+		r =rand.Float64()
+	}
+	
+	//special returns
+	if alpha < 0 || beta < 0 {
+		log.Fatalf("Error in RandBeta: alpha and beta must be greater than 0.  alpha: %f. beta: %f.", alpha, beta)
+	}
+	if math.IsInf(alpha, 1) && math.IsInf(beta, 1) {
+		return 0.5
+	}
+	if alpha == 0 && beta == 0 {
+		return 1.0
+	}
+	if math.IsInf(alpha, 1) || beta == 0 {
+		return 1.0
+	}
+	if math.IsInf(beta, 1) || alpha == 0 {
+		return 0.0
+	}
+
+	var sum, a, b, bet, delta, k1, k2, s, t, u1, u2, v, w, y, z float64
+	var i int
+	a = MinFloat64(alpha, beta)
```

Comment:
These two lines look they are are to make a the smaller of the two and b the bigger of the two.  With a single comparison we could do if a <= b { a = alpha, b = beta } else { a = beta, b = alpha }